src/App.vue

```diff
@@ -324,23 +389,119 @@ export default {
       }
     },
 
+    // FUTURE: update this when API returns New Incorporation task (#3102)
+    /** Gets NR data from Namex API. */
+    async getNrData (): Promise<any> {
+      await this.intializeNameXToken()
+      return this.queryNameRequest(this.nrNumber)
+    },
+
+    storeNrData (data: any): void {
+      if (!this.isNrValid(data)) {
+        this.nameRequestInvalidDialog = true
+        throw new Error('Invalid NR data')
+      }
+      if (!this.isNrConsumable(data)) {
+        this.nameRequestInvalidDialog = true
+        throw new Error('NR not consumable')
+      }
+
+      // FOR DEBUGGING ONLY
+      if (new Date(data.expirationDate) < new Date()) {
+        data.expirationDate = new Date() // today
+      }
```

Comment:
FYI, this is because the expiration data in our test NRs is several months ago :]

(Now updated to work with unit tests/hard-coded data.)

Reply:
Excellent!

Reply:
Curious, do we know how long the expiration date will be in the end?

Reply:
I think it's 60 days.

What needs to be clarified is how the object's 'expiration date' field relates to its 'state' field. For example, our test NRs are expired (eg, "expirationDate":"Thu, 21 Nov 2019 08:00:00 GMT") but their state is still APPROVED instead EXPIRED. Maybe the Namex API will update this when it responds with the data, or maybe there's a nightly job updating data... I will follow up with Lorna on this.